apps/qubit/modules/user/actions/loginAction.class.php

```diff
@@ -25,7 +25,8 @@ public function execute($request)
     $this->form->getValidatorSchema()->setOption('allow_extra_fields', true);
 
     // Redirect to @homepage if the user is already authenticated
-    if ($this->context->user->isAuthenticated())
+    // or the read only mode is enabled
+    if ($this->context->user->isAuthenticated() || Qubit::isReadOnly())
```

Comment:
Wouldn't make more sense to forward the user to the unauthorized page if the app is in read only mode?